2_dataCleaning/DataCleaning_Code.Rmd

```diff
@@ -206,9 +222,10 @@ You may also want to recalculate or rescale some variables. For example, we can
 
 ```{r}
 
-# calculate Z scores
-mydata$NeuroticismZ <- scale(mydata$Neuroticism, center = TRUE, scale = TRUE)
-hist(mydata$NeuroticismZ)
+# calculate Z scores using mutate ()  and scale() 
+mydata <- mydata %>%
+  mutate(NeuroticismZ = scale(Neuroticism, center = TRUE, scale = TRUE))
+
 
 # one way to average multiple columns
 mydata$Average <- rowMeans(mydata[c("T1", "T2", "T3", "T4")])
```

Comment:
calculating row means in dplyr::mutate() is a little wacky because it expects a matrix as a single arg, and doesn't flexibly take column names as args, so `mydata <- mydata %>% mutate(Average = rowMeans(T1, T2, T3, T4))` won't work.

here's a possible strategy to suggest:

`mydata <- mydata %>% mutate(Average = rowMeans(select(., T1, T2, T3, T4))` and then whatever columns you want have to go inside mutate()

I actually run into this issue enough times that I should write a wrapper function for rowMeans that behaves nicely inside of mutate(), like tidyRowMeans or something. I can put it in the cuPsychRTools package and we can load it in the tutorial?